common/FormattingUtils.js

```diff
@@ -31,10 +31,56 @@ export function formatWithMaxDecimals(num, decimalPlaces, roundUp) {
   const fullPrecisionFloat = BigNumber(num);
 
   // Convert back to BN to truncate any trailing 0s that the toFixed() output would print.
-  const fixedPrecisionFloat = BigNumber(fullPrecisionFloat.toFixed(decimalPlaces));
-  return fixedPrecisionFloat.toString();
-}
+  const fixedPrecisionFloat = BigNumber(fullPrecisionFloat).toFixed(decimalPlaces);
+  return fixedPrecisionFloat.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+};
 
-export function createFormatFunction(web3, numDisplayedDecimals) {
+const createFormatFunction = (web3, numDisplayedDecimals) => {
   return valInWei => formatWithMaxDecimals(formatWei(valInWei, web3), numDisplayedDecimals, false);
+};
+
+// generate a etherscan link prefix
+function createEtherscanLinkFromtx(web3, networkUtils) {
+  // Construct etherscan link based on network
+  // const networkId = web3.networkId;
+  const networkId = 1;
```

Comment:
hard code the networkID to 1 for now for testing. in future this will build the link based on the network the bot is running from

Reply:
We already have [code](https://github.com/UMAprotocol/protocol/blob/5711c55d75db38651c65b9b78ae134c66ea72502/sponsor-dapp-v2/src/lib/custom-hooks.js#L58-L65) that basically does this in the sponsor dapp.

Rather than trying to pull that code out or something, you import [this file](https://github.com/UMAprotocol/protocol/blob/master/common/PublicNetworks.js), and it should be pretty trivial to look up the correct etherscan prefix (or fall back to a default).

Reply:
right. check the commented out line on 45 that pulls the network id from the web3 instance. I am using the Public networks file already to get the base URL. I've refactored this such that this `FormattingUtils.js` pulls directly from `PublicNetworks.js` now and changed the default behaviour to return a public main net etherscan link.